frontend/projects/upgrade/src/app/core/error-event-bus/store/error-event-bus.reducer.ts

```diff
@@ -0,0 +1,28 @@
+import { createReducer, on } from '@ngrx/store';
+import * as ErrorActions from './error-event-bus.actions';
+import { SERVER_ERROR } from 'upgrade_types';
+
+export interface ErrorState {
```

Comment:
For each error, we will have a new attribute?

Reply:
Yeah, that's the idea. I had a different idea at first, which was basically to have just one error stream that all components could listen to on their own for errors of that type.

I think I ended up liking this better just so all the logic for creating and cleaning up errors could be kinda managed by this feature itself, minimal code for the component, which just needs to subscribe to that error type to catch it and use it like a boolean flag, and `clear` to mark that it was handled to remove it.